### PR TITLE
Authentication guard/some more alerts

### PR DESCRIPTION
**Pull request by dsgoers**

### CODE REVIEW

File: src/app/service/auth-guard/auth-guard.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import 'rxjs/add/operator/toPromise';
```
Comment (OzzieOrca) [inline src/app/service/auth-guard/auth-guard.service.spec.ts]: I don't see this being used. If you were having errors without it, maybe it should be included in the file where you convert the subscription to a promise.

Comment (dsgoers) [reply src/app/service/auth-guard/auth-guard.service.spec.ts]: If I don't include that import I get this error:  `Property 'toPromise' does not exist on type 'Observable<Response>'.`

Comment (OzzieOrca) [reply src/app/service/auth-guard/auth-guard.service.spec.ts]: I guess you can leave it for now. I could be wrong but I think it probably belongs in a different file. You could check if all your files that use `toPromise` have that import.

**Comment by dsgoers** [comment]
thanks!

**Pull request merged**